go/cli/cmd/kagent/main.go

```diff
@@ -29,8 +42,13 @@ func main() {
 
 	client := autogen_client.New(cfg.APIURL, cfg.WSURL)
 
-	// Attempt to connect to the server
-	_, err = client.GetVersion()
+	// Check server connection for commands passed as arguments
+	if len(os.Args) > 1 && os.Args[1] != "install" && os.Args[1] != "version" {
+		if err := checkServerConnection(client); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	}
```

Comment:
How does this work? We don't currently support passing in any args from the command line afaik? This won't actually run the `install` command, right?